### PR TITLE
Goguen ala cardano docs [WIP]

### PR DESCRIPTION
**Pull request by jbgi**

### CODE REVIEW

**Comment by jbgi** [comment]
@deepfire should we merge this for now?

File: deployments/configurator.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    imports = [ <module/parameters.nix> ];
+    node    = { inherit accessKeyId org region; };
+    cluster = { inherit deployerIP hostedZone; };
+  };
+
+  ## Universal resource logic:
+  resources.route53HostedZones = optionalAttrs (hostedZone != null) {
+    hostedZone = { config, ... }: {
+      inherit region accessKeyId;
+      name = "${hostedZone}";
```
Comment (cleverca22) [inline deployments/configurator.nix]: you can also just `name = hostedZone;`

File: goguen/secp256k1.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+stdenv.mkDerivation rec {
+  name = "secp256k1";
+  src  = getSrc name;
+
+  buildInputs = [ autoconf automake libtool ];
+
+  configurePhase = ''
+    ./autogen.sh
+    ./configure --enable-module-recovery --enable-module-ecdh --enable-experimental --prefix $lib
+  '';
```
Comment (cleverca22) [inline goguen/secp256k1.nix]:
```
buildInputs = [ autoreconfHook ];
configureFlags = [ "--enable-module-recovery" "--enable-module-ecdh" .... ];
```
that will run autoreconf for you

File: goguen/solidity.nix [hunk 3]
```diff
@@ -20,8 +20,11 @@ pkgs.stdenv.mkDerivation {
 
   patches = [ ./solidity.patch ];
 
-  preConfigure = ''
+  configurePhase = ''
     export NIX_CFLAGS_COMPILE="-Wno-error=maybe-uninitialized $NIX_CFLAGS_COMPILE"
```
Comment (cleverca22) [inline goguen/solidity.nix]: why not keep this in `preConfigure` ?

File: modules/exporer-remix-solidity-services.nix [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            };
+            "/remix/".proxyPass = "http://127.0.0.1:3000/";
+          };
+        };
+      };
+    };
+
+    users.users.eth = {
+      isNormalUser = true;
+      home = "/home/eth";
```
Comment (cleverca22) [inline modules/exporer-remix-solidity-services.nix]: `isNormalUser = true;` sets `home` for you, to the same value you already have

File: modules/mantis-faucet-service.nix [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    deployment.keys."mallet.json" = {
+      keyFile = <static> + "/mallet-${name}.json";
+      user = "mantis";
+      destDir = "/var/lib/keys";
+    };
+
+    users.users.mantis =
+        { isNormalUser = true;
+          home = "/home/mantis";
```
Comment (cleverca22) [inline modules/mantis-faucet-service.nix]: `isNormalUser` again

**Pull request closed without merging**